### ISSUE TITLE
Some sentences do not appear in the search results

### ISSUE THREAD

**Issue by Yorwba** [Tatoeba/tatoeba2#1952]
Wall thread: https://tatoeba.org/eng/wall/show_message/32577#message_32577

There are [1679 Mandarin sentences with audio](https://tatoeba.org/eng/audio/index/cmn), but [a search for them only returns a small percentage](https://tatoeba.org/eng/sentences/search?query=&from=cmn&to=und&orphans=no&unapproved=no&user=&tags=&list=&has_audio=yes&trans_filter=limit&trans_to=und&trans_link=&trans_user=&trans_orphan=&trans_unapproved=&trans_has_audio=&sort=relevance), though the number appears to be growing slowly.

Removing the filtering for sentences with audio as well as restrictions on orphaned and unapproved sentences [yields 450 results](https://tatoeba.org/eng/sentences/search?query=&from=cmn&to=und&orphans=&unapproved=&user=&tags=&list=&has_audio=&trans_filter=limit&trans_to=und&trans_link=&trans_user=&trans_orphan=&trans_unapproved=&trans_has_audio=&sort=relevance), despite there being [60910 Mandarin sentences in total](https://tatoeba.org/eng/Sentences/show_all_in/cmn/none).

This also applies to Abkhaz, with the search only finding [25 sentences](https://tatoeba.org/eng/sentences/search?query=&from=abk&to=none&orphans=&unapproved=&user=&tags=&list=&has_audio=&trans_filter=limit&trans_to=und&trans_link=&trans_user=&trans_orphan=&trans_unapproved=&trans_has_audio=&sort=created) although [there should be 27](https://tatoeba.org/eng/Sentences/show_all_in/abk/none).

The lists are short enough that I could compare them and find a sentence that should appear, but didn't: [#3892187](https://tatoeba.org/sentences/show/3892187) (Иҭабуп!). Explicitly searching for that word [doesn't find anything](https://tatoeba.org/eng/sentences/search?query=%D0%98%D2%AD%D0%B0%D0%B1%D1%83%D0%BF&from=und&to=und). However, the same search [works on dev](https://dev.tatoeba.org/eng/sentences/search?query=%D0%98%D2%AD%D0%B0%D0%B1%D1%83%D0%BF&from=und&to=und).

That leads me to the question of how the Manticore configuration differs between dev and prod, if at all.

Maybe this is related to the index corruption in #1944, but the symptoms look very different.

**Comment by trang** [comment]
The configurations are the same on dev and prod. But there was a change in the config during the upgrade to Manticore 3: https://github.com/Tatoeba/tatoeba2/commit/c78d8afc8dedba842472991d54a4a11b5900b7ef

It feels like we have something wrong in the config related to these kill-lists, but it isn't noticeable on dev because we don't have much activity there.

There are now only 23 sentences found in the search where you found 25 sentences by the way.

**Comment by AndiPersti** [comment]
There is still a problem with the search.

[#5978209](https://tatoeba.org/eng/sentences/show/5978209) isn't found on [prod](https://tatoeba.org/eng/sentences/search?query=Y+tekove+heko&from=und&to=und&orphans=&unapproved=) **and** is neither found on [dev](https://dev.tatoeba.org/eng/sentences/search?query=Y+tekove+heko&from=und&to=und&orphans=&unapproved=) because for some reason [it doesn't exist there](https://dev.tatoeba.org/eng/sentences/show/5978209).

[#3230000](https://tatoeba.org/eng/sentences/show/3230000) isn't found on [prod](https://tatoeba.org/eng/sentences/search?query=%D0%9E%D1%87%CD%A0%D0%B5+%D0%BD%D0%B0%D1%88%D1%8C+%D0%B8%D0%B6%D1%94&from=und&to=und&orphans=&unapproved=) but is found on dev. The same for [#1980618](https://tatoeba.org/eng/sentences/show/1980618), [#950300](https://tatoeba.org/eng/sentences/show/950300), [#3239167](https://tatoeba.org/eng/sentences/show/3239167), [#380264](https://tatoeba.org/eng/sentences/show/380264), and [#1892592](https://tatoeba.org/eng/sentences/show/1892592).

**Comment by jiru** [comment]
> [#5978209](https://tatoeba.org/eng/sentences/show/5978209) isn't found on [prod](https://tatoeba.org/eng/sentences/search?query=Y+tekove+heko&from=und&to=und&orphans=&unapproved=)

This one is orphan, that’s why it doesn’t show up by default. You need to enable orphan sentences in the search criteria.

> **and** is neither found on [dev](https://dev.tatoeba.org/eng/sentences/search?query=Y+tekove+heko&from=und&to=und&orphans=&unapproved=) because for some reason [it doesn't exist there](https://dev.tatoeba.org/eng/sentences/show/5978209).

This is the expected behaviour. Dev is always outdated because we don’t update it unless we have something to test there.

> [#3230000](https://tatoeba.org/eng/sentences/show/3230000) isn't found on [prod](https://tatoeba.org/eng/sentences/search?query=%D0%9E%D1%87%CD%A0%D0%B5+%D0%BD%D0%B0%D1%88%D1%8C+%D0%B8%D0%B6%D1%94&from=und&to=und&orphans=&unapproved=) but is found on dev.

I confirm. It appears the index is okay and the sentence was just not included. Reindexing ORV solved the problem.

> The same for [#1980618](https://tatoeba.org/eng/sentences/show/1980618), [#950300](https://tatoeba.org/eng/sentences/show/950300), [#3239167](https://tatoeba.org/eng/sentences/show/3239167), [#380264](https://tatoeba.org/eng/sentences/show/380264), and [#1892592](https://tatoeba.org/eng/sentences/show/1892592).

Yeah, except for [#1980618](https://tatoeba.org/eng/sentences/show/1980618) and [#380264](https://tatoeba.org/eng/sentences/show/380264). They are correctly indexed now.

So reindexation seems to solve the problem. Maybe this bug is just a side effect of #1944.

**Comment by AndiPersti** [comment]
> This is the expected behaviour. Dev is always outdated because we don’t update it unless we have something to test there.

Yes, I understand that. But I just found it strange that it doesn't appear on Dev although there were logs for it and it's from 2017 (AFAIK the database on Dev is from May 2019).

> So reindexation seems to solve the problem. Maybe this bug is just a side effect of #1944.

Yes, it looks like reindexing the database is necessary. I've just retested Abkhaz and although Trang said that all 27 sentences were found after rebuilding the index in September, it again "lost" one sentence: [#4256204](https://tatoeba.org/deu/sentences/show/4256204). Searching for these words currently [doesn't find anything](https://tatoeba.org/deu/sentences/search?query=%D0%91%D0%B7%D0%B8%D0%B0%D0%BB%D0%B0+%D1%88%D3%99%D0%B0%D0%B0%D0%B1%D0%B5%D0%B8%D1%82+%D0%90%D0%B2%D0%B8%D0%BA%D0%B8%D0%BF%D0%B5%D0%B4%D0%B8%D0%B0%D1%85%D1%8C&from=abk&to=none&user=&orphans=&unapproved=&has_audio=&tags=&list=&native=&trans_filter=limit&trans_to=und&trans_link=&trans_user=&trans_orphan=&trans_unapproved=&trans_has_audio=&sort=relevance&sort_reverse=).
